### PR TITLE
Move ITT testing to its own test case

### PR DESCRIPTION
**Pull request by jingxu10**
Fixes https://github.com/pytorch/pytorch/pull/84848#discussion_r986329680

### CODE REVIEW

**Comment by slgong-fb** [comment]
@jingxu10 Thanks for quick patch. I added ciflow/trunk label for extended testing.

File: test/profiler/test_profiler.py
Comment (malfet) [inline test/profiler/test_profiler.py:133]: Sorry, but is there an indication that test if flaky on any of those platforms?

**Review by malfet — CHANGES_REQUESTED** [review]
Not sure I understand what bug one is trying to fix? `test_profiler` works fine as is right now, isn't it? is there a record of flaky failures?

Also, if you want to skip it on Windows/Mac, why not just decorate one method in the same class rather than introducing a whole new class?

File: test/profiler/test_profiler.py
Comment (jingxu10) [reply test/profiler/test_profiler.py:133]: Hi @malfet  I haven't tested ITT on Windows and MacOS. However, from Intel(R) VTune Profiler perspective, all Windows, Linux and MacOS are supported. Shall I remove these 2 skipifs to have the UT running on Windows and MacOS as well?

Comment (malfet) [reply test/profiler/test_profiler.py:133]: Yes, I think the only one needed here is `@unittest.skipIf(not HAS_PSUTIL, "Requires psutil to run")` and if test fails with ASAN, it should also be guarded against running on the platform

**Comment by jingxu10** [comment]
> Not sure I understand what bug one is trying to fix? `test_profiler` works fine as is right now, isn't it? is there a record of flaky failures?
> 
> Also, if you want to skip it on Windows/Mac, why not just decorate one method in the same class rather than introducing a whole new class?

In #84848 , the ITT ut was added in `TestProfilerCUDA`. @slgong-fb claimed his PR was blocked by `RuntimeError: Can't use ITT profiler - PyTorch was compiled without ITT`. However, the PR 84848 passed all tests before. This is the failured mentioned in the comment.
Indeed, the ITT ut is not good to be included in `TestProfilerCUDA`, so I moved it out into a new test class.

File: test/profiler/test_profiler.py
Comment (jingxu10) [reply test/profiler/test_profiler.py:133]: The ITT PR actually doesn't depend on psutil, shall we just skip if the device is not a x86 platform?
By the way, any idea how I can find the keyword of x86? I tried to search "IS_*" in test folder, but didn't get a candidate.

Comment (malfet) [reply test/profiler/test_profiler.py:133]: There is `IS_ARM64`, so you can do `@unittest.skip_if(IS_ARM64, "ITT is not available on ARM")`

Comment (jingxu10) [reply test/profiler/test_profiler.py:133]: Got it. Thanks.

Comment (jingxu10) [reply test/profiler/test_profiler.py]: Hi @malfet , Would you review my updated codes? Since the PR doesn't depend on psutils, I added 1 skipif to skip on ARM.

File: test/test_autograd.py
Comment (malfet) [inline test/test_autograd.py:8090]: Why are you removing onlyCPU decorator? Do you want to run it on CUDA?

File: test/test_itt.py
Comment (malfet) [inline test/test_itt.py:10]: Why this change is necessary?

**Comment by slgong-fb** [comment]
Thanks! @malfet and @jingxu10 for handling my complain so quick. The unit test failed on FB-internal. Not sure if `USE_ITT` flag is CI tests, but I believe the unit test itself was not in a good shape. Otherwise the unit test should pass without USE_ITT flag setup step.

**Review by malfet — COMMENTED** [review]
Not sure I understand what are you trying to achieve with this change? If tests were running previously, why sometimes restrict them, sometimes extend them (to GPUs)? Also, those extra changes to do align with PRs title.

If you want to restrict some tests to x86 only platform (or extend some to CUDA) can you please do it in separate PR, but keep this one limited to refactoring ITT specific tests from `TestProfiler. test_custom_module_input_op_ids` to `TestProfilerITT.test_custom_module_input_op_ids`

File: test/profiler/test_profiler.py
Comment (slgong-fb) [reply test/profiler/test_profiler.py]: `skipIf`,  not `skip_if`?

**Comment by jingxu10** [comment]
> Not sure I understand what are you trying to achieve with this change? If tests were running previously, why sometimes restrict them, sometimes extend them (to GPUs)? Also, those extra changes to do align with PRs title.
> 
> If you want to restrict some tests to x86 only platform (or extend some to CUDA) can you please do it in separate PR, but keep this one limited to refactoring ITT specific tests from `TestProfiler. test_custom_module_input_op_ids` to `TestProfilerITT.test_custom_module_input_op_ids`

Sure. I'll revert changes in other files.

**Comment by jingxu10** [comment]
> Thanks! @malfet and @jingxu10 for handling my complain so quick. The unit test failed on FB-internal. Not sure if `USE_ITT` flag is CI tests, but I believe the unit test itself was not in a good shape. Otherwise the unit test should pass without USE_ITT flag setup step.

I'll verify USE_ITT = OFF on my local environment.

**Review by malfet — APPROVED** [review]
Looks good to me

File: test/profiler/test_profiler.py
Comment (jingxu10) [reply test/profiler/test_profiler.py]: correct to skipIf

**Comment by jingxu10** [comment]
> Thanks! @malfet and @jingxu10 for handling my complain so quick. The unit test failed on FB-internal. Not sure if `USE_ITT` flag is CI tests, but I believe the unit test itself was not in a good shape. Otherwise the unit test should pass without USE_ITT flag setup step.

Once this PR is merged, I'll file another PR to solve the USE_ITT=OFF ut failure issue.

**Pull request closed without merging**
